PWGHF/Tasks/taskMcValidation.cxx

```diff
@@ -537,19 +487,19 @@ struct HfTaskMcValidationRec {
       if (track.has_mcParticle()) {
         auto particle = track.mcParticle(); // get corresponding MC particle to check origin
         if (checkAmbiguousTracksWithHfEventsOnly) {
-          auto mcCollision = particle.mcCollision_as<mcCollisionWithHFSignalInfo>();
+          auto mcCollision = particle.mcCollision_as<McCollisionWithHFSignalInfo>();
           if (!mcCollision.hasHFsignal()) {
             continue;
           }
         }
         auto origin = RecoDecay::getCharmHadronOrigin(particlesMC, particle, true);
         histTracks->Fill(origin, track.pt());
-        if (track.isAmbiguousTrack()) {
-          registry.fill(HIST("histAmbiguousTrackNumBC"), track.numBC());
-          registry.fill(HIST("histAmbiguousTrackNumCollisions"), track.ambiguousCollisionIndicesIds().size());
+        bool isAmbiguous = (track.compatibleCollIds().size() != 1);
```

Comment:
Is this the right condition? Do you not fill `histAmbiguousTrackZvtxRMS` with RMS = 0 for tracks that have no compatible collision?